src/test/ui/async-await/issues/issue-78938-async-block.stderr

```diff
@@ -0,0 +1,30 @@
+error[E0373]: async block may outlive the current function, but it borrows `room_ref`, which is owned by the current function
+  --> $DIR/issue-78938-async-block.rs:8:39
+   |
+LL |       let gameloop_handle = spawn(async {
+   |  _______________________________________^
+LL | |         game_loop(Arc::clone(&room_ref))
+   | |                               -------- `room_ref` is borrowed here
+LL | |     });
+   | |_____^ may outlive borrowed value `room_ref`
+   |
+note: function requires argument type to outlive `'static`
+  --> $DIR/issue-78938-async-block.rs:8:33
+   |
+LL |       let gameloop_handle = spawn(async {
+   |  _________________________________^
+LL | |         game_loop(Arc::clone(&room_ref))
+LL | |     });
+   | |_____^
+   = note: borrows cannot be held across a yield point, because the stack space of the current function is not preserved
+   = help: see https://rust-lang.github.io/async-book/03_async_await/01_chapter.html#awaiting-on-a-multithreaded-executor for more information
+help: to force the async block to take ownership of `room_ref` (and any other referenced variables), use the `move` keyword
+   |
+LL |     let gameloop_handle = spawn(async move {
+LL |         game_loop(Arc::clone(&room_ref))
+LL |     });
```

Comment:
I'll need to check the suggestion that generates this becuase it seems to me we don't really need to show the whole block for this and we could make it less verbose.

Reply:
Perhaps show only the line with `async`?

Reply:
Fixing that requires "only" to modify the following

https://github.com/rust-lang/rust/blob/afa74080410c1eba4cd23ddd8e2ac3d6ad1441e3/compiler/rustc_mir/src/borrow_check/diagnostics/conflict_errors.rs#L1287-L1301

The `args_span` should _only_ point at the `async {` and the suggested code sould _only_ be `"async move {"`, where now it is the whole thing.

If you _wish_ to tackle this in this PR, it'd be great, but I don't consider it a must have yet.

Reply:
No, I'll leave that for later or someone else.